core/src/main/scala/org/apache/spark/deploy/worker/DriverRunner.scala

```diff
@@ -53,9 +53,11 @@ private[deploy] class DriverRunner(
   @volatile private var killed = false
 
   // Populated once finished
-  private[worker] var finalState: Option[DriverState] = None
-  private[worker] var finalException: Option[Exception] = None
-  private var finalExitCode: Option[Int] = None
+  @volatile private[worker] var finalState: Option[DriverState] = None
```

Comment:
It's actually called by `WorkerPage.scala` which uses the `Option` to show that the driver is "running"